spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingServiceProperties.java

```diff
@@ -200,4 +200,24 @@ public String getBinder(String channelName) {
 		return bindings.get(channelName).getBinder();
 	}
 
+	/**
+	 * Return configuration properties as Map.
+	 * @return map of channel binding configuration properties.
+	 */
+	public Map<String, Object> asMapProperties() {
+		Map<String, Object> properties = new HashMap<>();
+		properties.put("instanceIndex", String.valueOf(getInstanceIndex()));
+		properties.put("instanceCount", String.valueOf(getInstanceCount()));
+		for (String name : getConsumerProperties().stringPropertyNames()) {
+			properties.put(name, getConsumerProperties());
+		}
+		for (String name : getProducerProperties().stringPropertyNames()) {
+			properties.put(name, getProducerProperties());
+		}
+		for (Map.Entry<String, BindingProperties> entry : getBindings().entrySet()) {
+			properties.put(entry.getKey(), entry.getValue());
+		}
+		return properties;
+	}
+
```

Comment:
I might have missed part of the conversation, but is this really what we want to track? 
Do we really want to track all the configuration/binding properties?
In XD we had other useful stuff like ip address, pid etc.
Can we capture the boot application/module name somehow?
I am concerned about name collision for a processor module - where we're adding both consumer and producer properties.
Finally, I think we need `toString()` on `BindingProperties` (if we include them) - the history is pretty useless in logs:

    SPRING_CLOUD_STREAM_HISTORY=[{input=org.springframework.cloud.stream.config.BindingProperties@19af3d5b, 
    output=org.springframework.cloud.stream.config.BindingProperties@3fe52e89, instanceIndex=0, instanceCount=1}]}]